### ISSUE TITLE
Unable to login on an Azure B2C enabled application with Facebook credentials

### ISSUE THREAD

**Issue by tvperez76** [AzureAD/microsoft-authentication-library-for-js#1073]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## Browser:
- [ ] Chrome version XX
- [ ] Firefox version XX
- [ ] IE version XX
- [ ] Edge version XX
- [x] Safari version 12

## Library version

<pre><code>
Library version: 1.2.0-beta.2 (and prior)

</code></pre>

## Current behavior
Whenever I try to log in to my web app, I get the following errors:
```
main-es2015.848f808….js:1 2 "Thu, 24 Oct 2019 19:58:48 GMT:1.2.0-beta.2-Info Returned from redirect url" false
main-es2015.848f808….js:1 1 "Thu, 24 Oct 2019 19:58:48 GMT:1.2.0-beta.2-Warning This overload for callback is deprecated - please change the format of the callbacks to a single callback as shown: (err: AuthError, response: AuthResponse)." false
main-es2015.848f808….js:1 2 "Thu, 24 Oct 2019 19:58:48 GMT:1.2.0-beta.2-Info Processing the callback from redirect response" false
main-es2015.848f808….js:1 2 "Thu, 24 Oct 2019 19:58:48 GMT:1.2.0-beta.2-Info State status:true; Request type:LOGIN" false
main-es2015.848f808….js:1 2 "Thu, 24 Oct 2019 19:58:48 GMT:1.2.0-beta.2-Info State is right" false
main-es2015.848f808….js:1 2 "Thu, 24 Oct 2019 19:58:48 GMT:1.2.0-beta.2-Info Fragment has id token" false
main-es2015.848f808….js:1 Token Object
main-es2015.848f808….js:1 2 "Thu, 24 Oct 2019 19:58:48 GMT:1.2.0-beta.2-Info isCallback will be deprecated in favor of urlContainsHash in MSAL.js v2.0." false
main-es2015.848f808….js:1 2 "Thu, 24 Oct 2019 19:58:51 GMT:1.2.0-beta.2-Info Add msal frame to document:msalRenewFramehttps://mlpexpress.onmicrosoft.com/webapi/common openid https://mlpexpress.onmicrosoft.com/webapi/user_impersonation" false
main-es2015.848f808….js:1 2 "Thu, 24 Oct 2019 19:58:51 GMT:1.2.0-beta.2-Info LoadFrame: msalRenewFramehttps://mlpexpress.onmicrosoft.com/webapi/common openid https://mlpexpress.onmicrosoft.com/webapi/user_impersonation" false
main-es2015.848f808….js:1 2 "Thu, 24 Oct 2019 19:58:51 GMT:1.2.0-beta.2-Info Add msal frame to document:msalRenewFramehttps://mlpexpress.onmicrosoft.com/webapi/common openid https://mlpexpress.onmicrosoft.com/webapi/user_impersonation" false
Refused to display 'https://m.facebook.com/dialog/oauth?client_id=958987680930672&redirect_uri=…TQ4YzZlMjZjIiwiVElEIjoiMzg4NTZhMDgtOTE3Mi00OTQzLWJlN2UtMTc4MDBhZjdlMGQ2In0' in a frame because it set 'X-Frame-Options' to 'DENY'.
26393main-es2015.848f808….js:1 0 "Thu, 24 Oct 2019 19:59:01 GMT:1.2.0-beta.2-Error Error when acquiring token for scopes: https://mlpexpress.onmicrosoft.com/webapi/common,openid,https://mlpexpress.onmicrosoft.com/webapi/user_impersonation - ClientAuthError: URL navigated to is https://login.microsoftonline.com/te/mlpexpress.onmicrosoft.com/b2c_1_mlpex…zations&client-request-id=d777b1ba-cb91-495b-886e-b2e3ec52fcb3&prompt=none, Token renewal operation failed due to timeout." false
11458main-es2015.848f808….js:1 0 "Thu, 24 Oct 2019 19:59:02 GMT:1.2.0-beta.2-Error Error when acquiring token for scopes: https://mlpexpress.onmicrosoft.com/webapi/common,openid,https://mlpexpress.onmicrosoft.com/webapi/user_impersonation - ClientAuthError: URL navigated to is https://login.microsoftonline.com/te/mlpexpress.onmicrosoft.com/b2c_1_mlpex…zations&client-request-id=d777b1ba-cb91-495b-886e-b2e3ec52fcb3&prompt=none, Token renewal operation failed due to timeout." false
```

## Expected behavior
Logging into my app without incident.

## Minimal reproduction of the problem with instructions
Navigate to [MLPExpress (dev)](https://mlpexpress.biz) on that has Safari 12 installed.  I can confirm that earlier versions work as expected.

**Comment by sameerag** [comment]
@tvperez76 Can you please try `msal 1.2.0-beta.3` in the same environment? We have added some more logging information that may help us to debug the timeouts. We will also debug from our end with Safari 12 to reproduce the issue.

**Comment by tvperez76** [comment]
Sure.  I will do so later today.

**Comment by tvperez76** [comment]
Here are the logs for library version 1.2.0-beta.3.

```main-es2015.4e16e0e….js:1 2 "Tue, 29 Oct 2019 03:22:52 GMT:1.2.0-beta.3-Info Returned from redirect url" false
main-es2015.4e16e0e….js:1 1 "Tue, 29 Oct 2019 03:22:52 GMT:1.2.0-beta.3-Warning This overload for callback is deprecated - please change the format of the callbacks to a single callback as shown: (err: AuthError, response: AuthResponse)." false
main-es2015.4e16e0e….js:1 2 "Tue, 29 Oct 2019 03:22:52 GMT:1.2.0-beta.3-Info Processing the callback from redirect response" false
main-es2015.4e16e0e….js:1 2 "Tue, 29 Oct 2019 03:22:52 GMT:1.2.0-beta.3-Info State status:true; Request type:LOGIN" false
main-es2015.4e16e0e….js:1 2 "Tue, 29 Oct 2019 03:22:52 GMT:1.2.0-beta.3-Info State is right" false
main-es2015.4e16e0e….js:1 2 "Tue, 29 Oct 2019 03:22:52 GMT:1.2.0-beta.3-Info Fragment has id token" false
main-es2015.4e16e0e….js:1 Token Object {uniqueId: "e4905626-6e40-4e9e-9e30-8defbec10c11", tenantId: undefined, tokenType: "id_token", idToken: I, idTokenClaims: Object…}
main-es2015.4e16e0e….js:1 2 "Tue, 29 Oct 2019 03:22:52 GMT:1.2.0-beta.3-Info isCallback will be deprecated in favor of urlContainsHash in MSAL.js v2.0." false
main-es2015.4e16e0e….js:1 2 "Tue, 29 Oct 2019 03:22:59 GMT:1.2.0-beta.3-Info Add msal frame to document:msalRenewFramehttps://mlpexpress.onmicrosoft.com/webapi/common openid https://mlpexpress.onmicrosoft.com/webapi/user_impersonation" false
main-es2015.4e16e0e….js:1 2 "Tue, 29 Oct 2019 03:22:59 GMT:1.2.0-beta.3-Info LoadFrame: msalRenewFramehttps://mlpexpress.onmicrosoft.com/webapi/common openid https://mlpexpress.onmicrosoft.com/webapi/user_impersonation" false
main-es2015.4e16e0e….js:1 2 "Tue, 29 Oct 2019 03:23:00 GMT:1.2.0-beta.3-Info Add msal frame to document:msalRenewFramehttps://mlpexpress.onmicrosoft.com/webapi/common openid https://mlpexpress.onmicrosoft.com/webapi/user_impersonation" false
Refused to display 'https://m.facebook.com/dialog/oauth?client_id=958987680930672&redirect_uri=…Dg1NTdiOTUwIiwiVElEIjoiODZmZjBlMTItZmFkYi00NmI5LTlmNGYtYzY2MzcxZDE0Nzc3In0' in a frame because it set 'X-Frame-Options' to 'DENY'.
4409main-es2015.4e16e0e….js:1 0 "Tue, 29 Oct 2019 03:23:09 GMT:1.2.0-beta.3-Error Error when acquiring token for scopes: https://mlpexpress.onmicrosoft.com/webapi/common,openid,https://mlpexpress.onmicrosoft.com/webapi/user_impersonation - ClientAuthError: URL navigated to is https://login.microsoftonline.com/te/mlpexpress.onmicrosoft.com/b2c_1_mlpex…zations&client-request-id=ea4a92a4-27b2-4e04-bec5-d07f40a1c619&prompt=none, Token renewal operation failed due to timeout." false
1578main-es2015.4e16e0e….js:1 0 "Tue, 29 Oct 2019 03:23:10 GMT:1.2.0-beta.3-Error Error when acquiring token for scopes: https://mlpexpress.onmicrosoft.com/webapi/common,openid,https://mlpexpress.onmicrosoft.com/webapi/user_impersonation - ClientAuthError: URL navigated to is https://login.microsoftonline.com/te/mlpexpress.onmicrosoft.com/b2c_1_mlpex…zations&client-request-id=ea4a92a4-27b2-4e04-bec5-d07f40a1c619&prompt=none, Token renewal operation failed due to timeout." false
```

**Comment by sameerag** [comment]
@tvperez76 Couple of more details: Can you help me understand the policy set in your sample? I see that you are making an `acquireTokenSilent` call but certain policies which need interaction are blocked on the same call by B2C service (eg: Password Reset). We are working with B2C dev team on documenting the usage of `msal js` per policy.

**Comment by tvperez76** [comment]
I’m not sure what you’re asking.  Are you requesting time see the login policy I’m using?  If so, I’m not sure how to provide this information.  I will see if I can provide an ARM template snippet of this.

**Comment by sameerag** [comment]
@tvperez76 Some of the B2C policies that requires user interaction like `password_reset` cannot call `acquireTokenSilent`. The B2C service blocks interactions in iframes and we create a hidden iframe for silent calls. This is expected behavior, we are working on consolidating all the use cases that would require this to add the guidance.

So the question is, does your policy need interaction but is it calling `acquireTokenSilent` to achieve the same?

**Comment by tvperez76** [comment]
In this case, I'm using the "Sign up and sign in" User Flow.  After logging in, are we to execute the aquireTokenRedirect()?  Because, the behavior I describes happen a loginRedirect() process has been completed.

**Comment by sameerag** [comment]
I think there could be combination of two issues in this case:
- Safari latest version cookie policies (I need to verify this)
- acquireTokenSilent being used for a use case where interaction is required

Can you please share a sample with me to reproduce at our end? You can email me if you do not want to publicly share it.

**Comment by jmckennon** [comment]
Closing this issue, as it's been some time and I'm assuming the issue has been resolved. Feel free to open a new issue if you have further questions.
